Haskell/Blarney/Netlist/Passes/ConstantFold.hs

```diff
@@ -89,9 +89,10 @@ evalConstNet n@Net{..} = case n of
   -- Mux
   NetP (Mux _ w) (Lit s:xs) -> modNet (Identity w) [xs !! fromInteger s]
   -- Registers
-  NetP (RegisterEn i w) [Lit 0, _] -> modNet (Const w i) []
+  NetP (RegisterEn Nothing w) [Lit 0, _] -> modNet (DontCare w) []
+  NetP (RegisterEn (Just i) w) [Lit 0, _] -> modNet (Const w i) []
   NetP (RegisterEn i w) [Lit 1, a] -> modNet (Register i w) [a]
-  NetP (Register i w) [Lit x] | i == x -> modNet (Const w i) []
+  NetP (Register (Just i) w) [Lit x] | i == x -> modNet (Const w i) []
```

Comment:
There is also a dont care case for this one with `Nothing` init val and `DontCare` input value